R/autopeek.R

```diff
@@ -0,0 +1,68 @@
+#' Get descriptive one-line summary of an object
+#'
+#' This function is created as a modification of [utils::str()] function. It is supposed to
+#' create more compacted yet informative summary about an object. It's default value of
+#' "icecream.peeking.function"
+#'
+#' @param object The object to be summarized.
+#' @param ... Other algorithms passed to methods.
+#'
+#' @details This is a generic function. Default method simply calls `utils::str` function.
+#'
+#' @return A string to be printed.
+#'
+#' @seealso [utils::str()] [ic_peek()]
+#' @export
+ic_autopeek <- function(object, ...) UseMethod("ic_autopeek")
+
+ic_autopeek.default <- str
+
+#' @importFrom glue glue
+#' @importFrom purrr map2_chr map_chr map_int detect_index
+#' @importFrom pillar obj_sum
+ic_autopeek_list_or_data.frame <- function(object,
+                                           max_summary_length = 70,
```

Comment:
Replacing `max_summary_length = 70` with `max_summary_length = getOption("width")` means we will query the current terminal width each time. I've played around quickly and it seems to work pretty well.

Following on from that I've been using `ic(iris)` as an example. It shrinks the line length until only the first column is displayed, but perhaps you could modify the code for `ic_autopeek_list_or_data.frame()` so the minimum it would show is `... data.frame [<nrow> x <ncol>]`, i.e. not display any column summaries? That would make it look good on even very slim terminals.

I.e. at present the least info that will be displayed is

``` r
ic(iris)
#> ℹ ic| `iris`: data.frame [150 x 5]: $'Sepal.Length': dbl [150], ...
```

So if you have a _very_ narrow terminal (this example is 41 chars wide) you get wrapping:

```
                                         | <- terminal window edge
> ic(iris)                               |                               
ℹ ic| `iris`: data.frame [150 x 5]: $'Sep|
al.Length': dbl [150], ...               |
```

Whereas I think it would look nicer in these circumstances to lose the column summaries altogether:

```
                                         | <- terminal window edge
> ic(iris)                               |                               
ℹ ic| `iris`: data.frame [150 x 5]:      |
```

What do you think?

Reply:
Hey Lewin, I'll answer this one on behalf of Dominik. I made a pull request to Dominik's branch where I did exactly what you described here and covered this with some tests on lists, so it should be solved once he finishes the breakfast and gets back to the computer.

Reply:
Hi, @ErdaradunGaztea, nice to see you there ;) I've reviewed your changes and merged them into my branch.